File: backend/PocApi/Models/Cliente.cs
using PocApi.Repositories;

namespace PocApi.Models;

public class Cliente : IEntity
{
    public int Id { get; set; }
    public string Nombre { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Telefono { get; set; } = string.Empty;
    public string Direccion { get; set; } = string.Empty;
}

public record ClienteDto(string Nombre, string Email, string Telefono, string Direccion);

File: frontend/poc_app/lib/models/cliente.dart
class Cliente {
  final int? id;
  final String nombre;
  final String email;
  final String telefono;
  final String direccion;

  Cliente({
    this.id,
    required this.nombre,
    required this.email,
    required this.telefono,
    required this.direccion,
  });

  factory Cliente.fromJson(Map<String, dynamic> json) => Cliente(
        id: json['id'] as int?,
        nombre: json['nombre'] as String,
        email: json['email'] as String,
        telefono: json['telefono'] as String,
        direccion: json['direccion'] as String,
      );

  Map<String, dynamic> toJson() => {
        'nombre': nombre,
        'email': email,
        'telefono': telefono,
        'direccion': direccion,
      };
}
